mlflow/store/tracking/file_store.py

```diff
@@ -1319,3 +1326,372 @@ def _read_helper(root, file_name, attempts_remaining=2):
                 return _read_helper(root, file_name, attempts_remaining - 1)
 
         return _read_helper(root, file_name, attempts_remaining=retries)
+
+    def start_trace(
+        self,
+        experiment_id: str,
+        timestamp_ms: int,
+        request_metadata: Dict[str, str],
+        tags: Dict[str, str],
+    ) -> TraceInfo:
+        """
+        Start an initial TraceInfo object in the backend store.
+
+        Args:
+            experiment_id: String id of the experiment for this run.
+            timestamp_ms: Start time of the trace, in milliseconds since the UNIX epoch.
+            request_metadata: Metadata of the trace.
+            tags: Tags of the trace.
+
+        Returns:
+            The created TraceInfo object.
+        """
+        request_id = request_metadata.get(SpanAttributeKey.REQUEST_ID)
+        if request_id is None:
+            raise MlflowException("Request ID is required to start a trace.", INTERNAL_ERROR)
+        _validate_experiment_id(experiment_id)
+        experiment_dir = self._get_experiment_path(
+            experiment_id, view_type=ViewType.ACTIVE_ONLY, assert_exists=True
+        )
+        mkdir(experiment_dir, FileStore.TRACES_FOLDER_NAME)
+        traces_dir = os.path.join(experiment_dir, FileStore.TRACES_FOLDER_NAME)
+        mkdir(traces_dir, request_id)
+        trace_dir = os.path.join(traces_dir, request_id)
+        trace_info = TraceInfo(
+            request_id=request_id,
+            experiment_id=experiment_id,
+            timestamp_ms=timestamp_ms,
+            execution_time_ms=None,
+            status=TraceStatus.IN_PROGRESS,
+            request_metadata=request_metadata,
+            tags=tags,
+        )
+        self._save_trace_info(trace_info, trace_dir)
+        return trace_info
+
+    def _save_trace_info(self, trace_info: TraceInfo, trace_dir, overwrite=False):
+        """
+        TraceInfo is saved into `traces` folder under the experiment, each trace
+        is saved in the folder named by its request_id.
+        `request_metadata` and `tags` folder store their key-value pairs such that each
+        key is the file name, and value is written as the string value.
+        Detailed directories structure is as below:
+        | - experiment_id
+        |   - traces
+        |     - request_id1
+        |       - meta.yaml
+        |       - request_metadata
+        |       - tags
+        |     - request_id2
+        |     - ...
+        |   - run_ids ...
+        """
+        # Save basic trace info to META_DATA_FILE_NAME
+        trace_info_dict = self._convert_trace_info_to_dict(trace_info)
+        write_yaml(trace_dir, FileStore.META_DATA_FILE_NAME, trace_info_dict, overwrite=overwrite)
```

Comment:
Is there a particular reason we store TraceInfo as yaml? If not we may want to use json to avoid having multiple serialization format.

Reply:
There's no difference I think, as we already have helper functions for writing & reading yaml here (run meta.yaml file) so I reuse it

Reply:
I'm actually not sure why we chose yaml to store metadata for runs and experiments. yaml makes sense if users manually edit the metadata file but they never do.

Reply:
Aligning with what runs and experiments do sounds good to me tho.